packages/core/src/auth/auth.ts

```diff
@@ -946,24 +946,13 @@ export class Auth implements AuthService, ConnectionManager {
 
     // Used by Amazon Q to delete connection status & scope when this deletion is made by AWS Toolkit
     // NO event should be emitted from this deletion
-    public async deletionConnectionCallback(id: string) {
+    // Do not actually perform the delete because toolkit has done the deletion
+    // Delete the momento states only.
+    public async onDeleteConnection(id: string) {
         const profile = this.store.getProfile(id)
-        // it is possible the connection was already deleted
-        // but was still requested to be deleted. We pretend
-        // we deleted it and continue as normal
         if (profile) {
```

Comment:
when debugging this, I noticed that this if statement never caught because in Q's `this.store`, the connection was already gone. It's like the toolkit is updating the store for Q, which it shouldn't have access to. Haven't narrowed down why this is the case, but I am surprised that this works given that.

Reply:
I ran a debugger and did not encounter the "connection already gone" issue after I restarted my VSC.

Reply:
:o